### ISSUE TITLE
LZ plays the same fuseki every single time.  

### ISSUE THREAD

**Issue by anonymousAwesome** [gcp/leela-zero#1165]
Black plays two corners at 3-4.  White plays two corners at 4-4.  Black and white play [the flower joseki](https://i.imgur.com/bB5zYSj.png) on white's corners.  Black jumps out two spaces from its corner stone.  Sometimes the order is mixed up a little, but it's the same fuseki every.  Single.  Time.  

Is that acceptable behavior, or are we supposed to be seeing more variation in the games?  Looking at [Alphago Zero's 20-block training records](http://www.alphago-games.com/#agzero_vs_agzero_20) I noticed that the first few moves in each record have more variety than LZ, even when the records are only 10-75 ELO apart.

Should we be worried about this?

See also: [this reddit post made a month ago commenting on the same oddity.](https://www.reddit.com/r/cbaduk/comments/832si0/why_are_all_the_self_playing_games_still_so_wierd/)

**Comment by PhilipFRipper** [comment]
You're looking at the match games?  Those are just for evaluation of strength, not for generating games for training.  So they have less randomness.  The self play games are what you want to look at for variety.  They have more randomness forced into their early moves and are what's used for training up networks.

**Comment by anonymousAwesome** [comment]
Yeah, I realized that about a minute ago. T_T Still, isn't the only difference between the two the number of playouts? Won't a LZ at full playouts stil try to play the same joseki every time?

**Comment by anonymousAwesome** [comment]
Regarding your edit: try to have some empathy, all right?  I didn't know, or at least I misunderstood the situation, and you telling me "this is basic knowledge" comes across as, "you should already just know this stuff" and makes me feel stupid for not already knowing what is obvious to you.  

Different people have different levels of engagement.  I read posts on the cbaduk subreddit and skim the match games as they update, and that's about it.  I didn't realize there was a minimum technical requirement necessary to be involved in this community.

**Comment by grolich** [comment]
the difference is a lot more than number of playouts: 
self play games are meant to generate more training data and force/encourage LZ to explore more plays to learn from, so there is a certain amount of randomness added to the search/network output all game long.

in match games no such randomness is present.
Well, there is one source of randomness still there - each position is fed to the network reflected around a random symmetry, and since the net isn't perfect, it will often treat these as slightly different (That is the way AGZ was made, and the way this project was build a while ago, if there has been a change concerning that, I am not aware of it yet. Will check to make sure though).

So evaluation games should be a lot more similar than the self play games (if we were doing the AZ way, we wouldn't even be using symmetries, and evaluation matches would not be possible at all, as there would only be 2 possible games... that's a very good reason why Deep Mind didn't use eval matches in their AZ project).

**Comment by anonymousAwesome** [comment]
Okay, training games have fewer MCTS playouts and extra randomness, while match games have more playouts and don't have the extra randomness.  I get that much.

But then why does AZ have multiple fuseki at full playouts while LZ doesn't?

**Comment by PhilipFRipper** [comment]
Github isn't reddit or facebook.  I shouldn't be posting here either.  It's for people to coordinate their programming.

**Comment by carljohanr** [comment]
AZ also had relatively little variation in the self-play games in fuseki, as you can see if you download the games or watch the Redmond series. 

From an observer perspective it would be interesting to force LZ to start from X common fusekis in the match play, but it seems this is not required to create a strong program (and would violate the Zero, if it was actually part of the training).

**Comment by anonymousAwesome** [comment]
Ah, crap.  You're right.  I'm looking at the first 7 moves of the first twelve games of AZ vs AZ, full-strength, and it appears AZ only has two fuseki that it ever plays.

@gcp, feel free to close this issue.

**Comment by jkiliani** [comment]
@anonymousAwesome If you are the one who opens a Github thread, you can also close it, in addition to the repository owner who can close any issue.
